extensions/sql-database-projects/src/tools/netcoreTool.ts

```diff
@@ -96,13 +111,63 @@ export class NetCoreTool {
 		return undefined;
 	}
 
+	/**
+	 * This function checks if the installed dotnet version is 3.1 or greater.
+	 * Versions lower than 3.1 aren't supported for building projects.
+	 * Returns: True if installed dotnet version is supported, false otherwise.
+	 * 			Undefined if dotnet isn't installed.
+	 */
+	private async isNetCoreVersionSupported(): Promise<boolean | undefined> {
+		try {
+			const spawn = child_process.spawn;
+			let child: child_process.ChildProcessWithoutNullStreams;
+			let isSupported: boolean | undefined = undefined;
+
+			child = spawn('dotnet --version', [], {
+				shell: true
+			});
+
+			child.stdout.on('data', (data) => {
+				this.netCoreSdkInstalledVersion = String(data);
+			});
+
+			isSupported = await this.onExit(child);
+			if (!isSupported) {
+				this.netCoreError = netCoreFindError.netCoreVersionNotSupported;
+			}
+			return isSupported;
+		} catch (err) {
+			console.log(err);
+			return undefined;
+		}
+	}
+
+	private onExit(childProcess: child_process.ChildProcess): Promise<boolean | undefined> {
+		return new Promise((resolve, reject) => {
+			childProcess.on('exit', () => {
+				let isSupported: boolean = false;
+				if (semver.gte(this.netCoreSdkInstalledVersion!, '3.1.0')) {
```

Comment:
You should avoid having magic numbers like this - it makes it difficult to update/find later on. Would suggest using a single const value and then making any comments which reference this specific version instead just say something like "min required version"